File: fsharp/cars-assemble/CarsAssemble.fs
module CarsAssemble

let successRate (speed: int): float =
    if speed < 1 then 0.0
    elif speed < 5 then 1.0
    elif speed < 9 then 0.90
    elif speed < 10 then 0.80
    else 0.77

let productionRatePerHour (speed: int): float =
    successRate speed * float (speed * 221)

let workingItemsPerMinute (speed: int): int =
    int (productionRatePerHour speed) / 60

File: java/cars-assemble/src/main/java/CarsAssemble.java
public class CarsAssemble {

    private double successRate(int speed) {
        double successRate;

        if (speed < 5)
            successRate = 1.00;
        else if (speed < 9)
            successRate = 0.90;
        else if (speed < 10)
            successRate = 0.80;
        else
            successRate = 0.77;

        return successRate;
    }

    public double productionRatePerHour(int speed) {
        final double BASE_RATE = 221.0;

        return speed * BASE_RATE * successRate(speed);
    }

    public int workingItemsPerMinute(int speed) {
        final int MINUTSPERHOUR = 60;

        return (int) productionRatePerHour(speed) / MINUTSPERHOUR;
    }
}
